app/lib/features/tasks/widgets/task_entry.dart

```diff
@@ -0,0 +1,139 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:jiffy/jiffy.dart';
+
+final taskCommentsProvider =
+    FutureProvider.autoDispose.family<CommentsManager, Task>((ref, t) async {
+  return await t.comments();
+});
+
+class TaskEntry extends ConsumerWidget {
+  final Task task;
+  const TaskEntry({Key? key, required this.task}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final List<Widget> extraInfo = [];
+    final dueDate = task.utcDueRfc3339();
+    final isDone = task.isDone();
+    if (!isDone) {
+      if (dueDate != null) {
+        final due = Jiffy.parse(dueDate);
+        final now = Jiffy.now();
+        if (due.isBefore(now)) {
+          extraInfo.add(
+            Padding(
+              padding: const EdgeInsets.only(left: 3),
+              child: Tooltip(
+                message: due.format(),
+                child: Text(
+                  due.fromNow(),
+                  style: isDone
+                      ? null
+                      : Theme.of(context).textTheme.bodySmall!.copyWith(
+                            color: Theme.of(context).colorScheme.taskOverdueFG,
+                          ),
+                ),
+              ),
+            ),
+          );
+        } else {
+          // FIXME: HL today, tomorrow
+          extraInfo.add(
+            Padding(
+              padding: const EdgeInsets.only(left: 3),
+              child: Text(
+                due.fromNow(),
+                style: Theme.of(context).textTheme.bodySmall,
+              ),
+            ),
+          );
+        }
+      }
+    }
+    final description = task.description();
+    if (description != null) {
+      extraInfo.add(
+        Tooltip(
+          message: description.body(),
+          child: const Icon(Atlas.document_thin),
+        ),
+      );
+    }
+    extraInfo.add(
+      Consumer(
+        builder: (context, ref, child) =>
+            ref.watch(taskCommentsProvider(task)).when(
+                  data: (commentsManager) {
+                    if (!commentsManager.hasComments()) {
+                      return const SizedBox.shrink();
+                    }
+                    return Padding(
+                      padding: const EdgeInsets.only(left: 3),
+                      child: Wrap(
+                        children: [
+                          const Icon(Atlas.comment_thin),
+                          Text(
+                            commentsManager.commentsCount().toString(),
+                            style: Theme.of(context).textTheme.bodySmall,
+                          ),
+                        ],
+                      ),
+                    );
+                  },
+                  error: (e, s) => Text('loading comments failed: $e'),
+                  loading: () => const SizedBox.shrink(),
+                ),
+      ),
+    );
+    final assignees = task.assignees();
+    if (assignees.isNotEmpty) {}
```

Comment:
The purpose of this being here isn't clear to me. We do want to show something if its not empty???

Reply:
ah, that's old stuff. Gotta remove it.